src/librustc_typeck/check/method/probe.rs

```diff
@@ -1404,15 +1404,41 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                             // `potentially_unsatisfied_predicates`.
                             return ProbeResult::NoMatch;
                         } else {
-                            // Some nested subobligation of this predicate
-                            // failed.
-                            //
-                            // FIXME: try to find the exact nested subobligation
-                            // and point at it rather than reporting the entire
-                            // trait-ref?
-                            result = ProbeResult::NoMatch;
-                            let trait_ref = self.resolve_vars_if_possible(&trait_ref);
-                            possibly_unsatisfied_predicates.push(trait_ref);
+                            self.probe(|_| {
```

Comment:
Are you sure this results in no functional language changes?

Reply:
There are no functional issues. If there's a mix of trait and projection predicates, and the underlying divergence is due to a projection the output is not as clear as it should be, but I am working on expanding it.

Reply:
To clarify, the changes only affect diagnostic output. `possibly_unsatisfied_predicates` isn't used for anything else.